tests/vm.rs

```diff
@@ -759,6 +760,28 @@ string.slice s 1 (string.length s)
 }
 
 #[test]
+fn out_of_memory() {
+    let _ = ::env_logger::init();
+    let vm = make_vm();
+    vm.set_memory_limit(10);
+    let result = Compiler::new()
+        .run_expr::<Generic<A>>(&vm, "example", r#" [1, 2, 3, 4] "#);
+    match result {
+        // FIXME This should just need to match on the explicit out of memory error
+        Err(Error::VM(VMError::OutOfMemory { limit: 10, .. })) => (),
+        Err(Error::VM(VMError::Message(msg))) => assert!(msg.starts_with("Thread is out of memory: Limit 10"), "{}", msg),
+        Err(Error::Macro(errors)) => {
+            match *errors.errors[0].downcast_ref::<Error>().unwrap() {
+                Error::VM(VMError::OutOfMemory { limit: 10, .. }) => (),
+                _ => panic!("Unexpected error {:?}", errors.errors[0]),
+            }
+        }
+        Err(err) => panic!("Unexpected error `{:?}`", err),
+        Ok(_) => panic!("Expected an error"),
+    }
+}
```

Comment:
Yeah, it was made a bit more bulletproof than it needs to. Ideally I think that macro errors should be collapsed so that vm errors have less possible representations.